src/api/accounts.rs

```diff
@@ -0,0 +1,515 @@
+//! Partial implementation of the `Accounts` namespace.
+
+use crate::api::{Namespace, Web3};
+use crate::error::Error;
+use crate::helpers::CallFuture;
+use crate::types::{Address, Bytes, SignedData, SignedTransaction, TransactionParameters, H256, U256};
+use crate::Transport;
+use ethsign::{Error as EthsignError, SecretKey, Signature};
+use futures::future::{self, Either, FutureResult, Join3};
+use futures::{Async, Future, Poll};
+use parity_crypto::Keccak256;
+use rlp::RlpStream;
+
+/// Transaction base fee, this will be used as the default transaction fee in
+/// the case none is specified for signing.
+pub const TRANSACTION_BASE_GAS_FEE: U256 = U256([21000, 0, 0, 0]);
+
+/// `Accounts` namespace
+#[derive(Debug, Clone)]
+pub struct Accounts<T> {
+    transport: T,
+}
+
+impl<T: Transport> Namespace<T> for Accounts<T> {
+    fn new(transport: T) -> Self
+    where
+        Self: Sized,
+    {
+        Accounts { transport }
+    }
+
+    fn transport(&self) -> &T {
+        &self.transport
+    }
+}
+
+impl<T: Transport> Accounts<T> {
+    /// Gets the parent `web3` namespace
+    fn web3(&self) -> Web3<T> {
+        Web3::new(self.transport.clone())
+    }
+
+    /// Signs an Ethereum transaction with a given private key.
+    pub fn sign_transaction(&self, tx: TransactionParameters, key: &SecretKey) -> SignTransactionFuture<T> {
```

Comment:
I chose to implement a future type here since I didn't see `impl Future` used anywhere else. Also I don't think the use of `async`/`await` would be good here since this crate uses futures 0.1 which require a compat layer to work with `async`/`await`.

Reply:
I think it's appropriate to port to the async/.await notation as a separate task, as that would ideally just rewrite 90% of the async stuff

Reply:
Makes sense!